tests/test_suites/GaneshaTests/test_nfs_ganesha_file_corruption_on_master_failover.sh

```diff
@@ -68,44 +73,29 @@ EXPORT {
 }
 EOF
 
+# Create a file for testing with checksum
+head -c 3G /dev/random | tee "${TEMP_DIR}/test_file" > /dev/null
+
 sudo /usr/bin/ganesha.nfsd -f "${info[mount0]}/ganesha.conf"
 assert_eventually 'showmount -e localhost'
 
 sudo mount -vvvv localhost:/ "${TEMP_DIR}/mnt/ganesha"
 
-# Create a file for testing with checksum
-head -c 3G /dev/random | tee "${TEMP_DIR}/test_file" > /dev/null
-
 # Restart master server after 15 seconds
 (
 	sleep 15
 	assert_success saunafs_master_daemon restart
 ) &
 
 # Wait for Grace period so NFS Ganesha server will be ready
-sleep 5
+sleep 10
 
-# Try to copy the file after master restart
-while true; do
-	cp "${TEMP_DIR}/test_file" "$TEMP_DIR/mnt/ganesha/test_file" && break
-	echo "Unable to copy test_file through NFS, retrying in 5 seconds..."
-	sleep 5
-done
+# Copy the file after master restart
+cp "${TEMP_DIR}/test_file" "$TEMP_DIR/mnt/ganesha/test_file"
 
 # Get checksums
 checksum1=$(get_checksum "${TEMP_DIR}/test_file")
-
-# To get the checksum of the file from NFS could require several retries in case
-# NFS mount will not be available after restarting master server
-while true; do
-	checksum2=$(get_checksum "${TEMP_DIR}/mnt/ganesha/test_file")
-	if [ "${checksum2}" != "" ]; then
-		break
-	fi
-	echo "Unable to get checksums through NFS, retrying in 5 seconds..."
-	sleep 5
-done
-
+checksum2=$(get_checksum_with_direct_io "${TEMP_DIR}/mnt/ganesha/test_file")
```

Comment:
Previous logic was considering a possible failure when trying to read a file through NFS. How are you handling such failure with this logic?